console/src/main/java/org/georchestra/console/ws/backoffice/users/GDPRAccountWorker.java

```diff
@@ -0,0 +1,375 @@
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Throwables;
+
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Cleanup;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * GDPR (EU <a href="https://eugdpr.org/">General Data Protection
+ * Regulation</a>) compliance data gatherer and mangler.
+ * <p>
+ * For a given account, all it's GDPR sensitive data is gathered as a zip file
+ * bundle and returned as a {@link Resource}.
+ *
+ */
+@Slf4j
+public class GDPRAccountWorker {
+
+    private static final DateTimeFormatter FILENAME_DATE_FORMAT = DateTimeFormatter.ISO_LOCAL_DATE;
+    private static final DateTimeFormatter CONTENT_DATE_FORMAT = DateTimeFormatter.ISO_DATE_TIME;
+
+    private @Autowired @Setter(AccessLevel.PACKAGE) AccountGDPRDao accountGDPRDao;
+    private @Autowired @Setter(AccessLevel.PACKAGE) UserInfoExporter userInfoExporter;
+    private @Autowired @Setter(AccessLevel.PACKAGE) AccountDao accountDao;
+
+    public static @Value @Builder class DeletedAccountSummary {
+        private String accountId;
+        private int metadataRecords;
+        private int extractorRecords;
+        private int geodocsRecords;
+        private int ogcStatsRecords;
+    }
+
+    public @EventListener void handleAccountDeleted(@NonNull AccountDeleted event) {
+        log.info("GDPR: deleting or obfuscating all activity records related to the deleted account {}",
+                event.getAccount().getUid());
+
+        try {
+            DeletedAccountSummary summary = deleteAccountRecords(event.getAccount());
+            log.info("GDPR: {}", summary);
+        } catch (DataServiceException e) {
+            log.error("Error deleting activity records for account {}", event.getAccount().getUid(), e);
+        }
+    }
+
+    public DeletedAccountSummary deleteAccountRecords(@NonNull Account account) throws DataServiceException {
+        DeletedRecords recs = accountGDPRDao.deleteAccountRecords(account);
+        return DeletedAccountSummary.builder()//
+                .accountId(recs.getAccountId())//
+                .metadataRecords(recs.getMetadataRecords())//
+                .extractorRecords(recs.getExtractorRecords())//
+                .geodocsRecords(recs.getGeodocsRecords())//
+                .ogcStatsRecords(recs.getOgcStatsRecords())//
+                .build();
+    }
+
+    /**
+     * Creates a ZIP file resource containing all downloadable data for the given
+     * user.
+     * <p>
+     * Once the returned resource has been dealt with, {@link #dispose(Resource)}
+     * shall be called in order to clean up any temporary content it may be holding
+     * up.
+     * 
+     * @throws DataServiceException
+     * @throws NameNotFoundException
+     * @throws IOException
+     */
+    public Resource generateUserData(@NonNull String userId)
```

Comment:
Between this method and the one https://github.com/georchestra/georchestra/pull/2613/files#diff-e5db7a86e07c3ec439f58ec54dff232eR119, wondering which one should be private / protected.

Reply:
Making one package private and `@VisibleForTesting`, and leaving the one called by the controller public, nice catch.

Reply:
a143c014a01248bd0b1ffee20b03ef5b0c943c0c